UnityProject/Assets/Scripts/Core/SecureStuff/SecureMapsSaver.cs

```diff
@@ -68,81 +87,557 @@ public virtual void AddID(string ToAdd)
-							continue;
+							Field.SetValue(Object, GetGameObjectPath(ModField.Data, IPopulateIDRelation));
 						}
+						else
+						{
+							IPopulateIDRelation.FlagSaveKey(RootID, root, ModField);
+						}
+					}
+				}
+
+				if (Field.FieldType.IsSubclassOf(typeof(ScriptableObject)))
+				{
+					var SO = IPopulateIDRelation.ObjectsFromForeverID(ModField.Data, Field.FieldType);
+					Field.SetValue(Object, SO);
+					return;
+				}
+
+				//if Field is a class and is not related to unity engine.object Serialise it
+				if (Field.FieldType.IsSubclassOf(typeof(UnityEngine.Object))) return;
+
+
+				if (Field.FieldType.IsGenericType == false)
+				{
+					ProcessIndividualField(RootID, root, Field.GetValue(Object), ModField, IPopulateIDRelation, AllLoaded, AdditionalJumps);
+					return;
+				}
+			}
+
+
+			if (Field.FieldType.IsGenericType &&
+			    Field.FieldType.GetGenericTypeDefinition() == typeof(Dictionary<,>))
+				return; //skipping all dictionaries For now
+			if (Field.FieldType == typeof(System.Action)) return;
+			if (Field.FieldType.BaseType == typeof(UnityEngine.Events.UnityEventBase))
+				return; //TODO Handle separately Since it is same as Object references
+
+
+			var MonoSet = Field.GetValue(Object);
+			if (MonoSet == null) return;
+
+			if (Field.FieldType.IsGenericType && typeof(IEnumerable).IsAssignableFrom(Field.FieldType) &&
+			    Field.FieldType.GetGenericArguments()[0].IsValueType) //UnityEventBase Handle differently
+			{
+				ListHandle(RootID, root, Field, Object, ModField, Index, IPopulateIDRelation);
+			}
+			else
+			{
+				if (Field.FieldType.IsGenericType) return; //Unity editor can't handle this currently so same Functionality
+				Field.SetValue(Object, Librarian.Page.DeSerialiseValue(ModField.Data, Field.FieldType));
+			}
+		}
+
+		private static void LoadDatarecursive(string RootID,  Component root, object Object, HashSet<FieldData> IndividualObject,
+			IPopulateIDRelation IPopulateIDRelation,
+			bool AllLoaded = false, bool IsServer = true) //Has to be Component to restrict it from being used in silly places
+		{
+			try
+			{
+				if (Object == null)
+				{
+					Loggy.LogError("oh....");
+				}
+
+				foreach (var ModField in IndividualObject)
+				{
+					try
+					{
+						ProcessIndividualField(RootID,root, Object, ModField, IPopulateIDRelation, AllLoaded);
+					}
+					catch (Exception e)
+					{
+						Loggy.LogError(e.ToString());
+					}
+				}
+			}
+			catch (Exception e)
+			{
+				Loggy.LogError(e.ToString());
+			}
+		}
+
+		public static void LoadData(string RootID,  Component Object, HashSet<FieldData> IndividualObject,
+			IPopulateIDRelation IPopulateIDRelation,
+			bool AllLoaded = false, bool IsServer =true) //Has to be Component to restrict it from being used in silly places
+		{
+			if (Object == null) return;
+			LoadDatarecursive(RootID,Object, Object, IndividualObject, IPopulateIDRelation, AllLoaded);
+		}
+
+		private static bool IsGoodClientField(FieldInfo Field)
```

Comment:
why

Reply:
well the reason why is you don't want overwrite synchronised stuff, since you want to use a Server staet for that